tests/test_backend.py

```diff
@@ -31,174 +31,166 @@ def reverse_key(key):
-        caches_setting["default"]["KEY_FUNCTION"] = "test_backend.make_key"
-        caches_setting["default"]["REVERSE_KEY_FUNCTION"] = "test_backend.reverse_key"
-        cm = override_settings(CACHES=caches_setting)
-        cm.enable()
-        self.addCleanup(cm.disable)
+    def test_keys(self, available_caches):
+        cache, other = available_caches
+        cache.set("a", "1")
+        other.set("b", "2")
+        keys = cache.keys("*")
+        assert "a" in keys
+        assert "b" not in keys
 
-        self.cache = caches["default"]
 
-    def tearDown(self):
-        self.cache.clear()
+def test_custom_key_function(settings):
+    caches_setting = copy.deepcopy(settings.CACHES)
+    caches_setting["default"]["KEY_FUNCTION"] = "test_backend.make_key"
+    caches_setting["default"]["REVERSE_KEY_FUNCTION"] = "test_backend.reverse_key"
+    settings.CACHES = caches_setting
+    cache = caches["default"]
 
-    def test_custom_key_function(self):
-        if isinstance(self.cache.client, ShardClient):
-            self.skipTest("ShardClient doesn't support get_client")
+    if isinstance(cache.client, ShardClient):
+        pytest.skip("ShardClient doesn't support get_client")
 
-        for key in ["foo-aa", "foo-ab", "foo-bb", "foo-bc"]:
-            self.cache.set(key, "foo")
+    for key in ["foo-aa", "foo-ab", "foo-bb", "foo-bc"]:
+        cache.set(key, "foo")
 
-        res = self.cache.delete_pattern("*foo-a*")
-        self.assertTrue(bool(res))
+    res = cache.delete_pattern("*foo-a*")
+    assert bool(res)
 
-        keys = self.cache.keys("foo*")
-        self.assertEqual(set(keys), {"foo-bb", "foo-bc"})
-        # ensure our custom function was actually called
-        self.assertEqual(
-            {k.decode() for k in self.cache.client.get_client(write=False).keys("*")},
-            {"#1#foo-bc", "#1#foo-bb"},
-        )
+    keys = cache.keys("foo*")
+    assert set(keys) == {"foo-bb", "foo-bc"}
+    # ensure our custom function was actually called
+    assert {k.decode() for k in cache.client.get_client(write=False).keys("*")} == (
+        {"#1#foo-bc", "#1#foo-bb"}
+    )
+    cache.clear()
 
 
-class DjangoRedisCacheTests(unittest.TestCase):
-    def setUp(self):
-        self.cache = cache
+@pytest.fixture
+def default_cache():
+    yield cache
+    cache.clear()
 
-    def tearDown(self):
-        self.cache.clear()
 
-    def test_setnx(self):
+class TestDjangoRedisCache:
+    def test_setnx(self, default_cache):
         # we should ensure there is no test_key_nx in redis
-        self.cache.delete("test_key_nx")
-        res = self.cache.get("test_key_nx")
-        self.assertIsNone(res)
+        default_cache.delete("test_key_nx")
+        res = default_cache.get("test_key_nx")
+        assert res is None
 
-        res = self.cache.set("test_key_nx", 1, nx=True)
-        self.assertTrue(res)
+        res = default_cache.set("test_key_nx", 1, nx=True)
+        assert res
```

Comment:
I prefer it to be more specific
```suggestion
        assert res == True
```